hash_practice/exercises.py

```diff
@@ -1,19 +1,48 @@
 
+from array import array
+from distutils.log import error
+import string
+
+
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    anagram_dict = {}
+    for sorted_word in strings:
+        a = tuple(sorted(sorted_word))
+        if a in anagram_dict:
+           anagram_dict[a].append(sorted_word) 
+        else:
+           anagram_dict[a] = [sorted_word]
+    return list(anagram_dict.values()) 
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
         Time Complexity: ?
         Space Complexity: ?

```

Comment:
Looks like the complexity calculations got missed here, but I haven't been dinging people for it.